framework/src/modules/interoperability/mainchain/commands/message_recovery.ts

```diff
@@ -41,129 +39,93 @@ export class MainchainMessageRecoveryCommand extends BaseInteroperabilityCommand
 	public schema = messageRecoveryParamsSchema;
 
 	public get name(): string {
-		return 'messageRecovery';
+		return COMMAND_NAME_MESSAGE_RECOVERY;
 	}
 
 	public async verify(
 		context: CommandVerifyContext<MessageRecoveryParams>,
 	): Promise<VerificationResult> {
 		const {
-			params: { chainID, idxs, crossChainMessages, siblingHashes },
+			params: { chainID, crossChainMessages },
 		} = context;
-		const chainIdAsBuffer = chainID;
-		let terminatedChainOutboxAccount: TerminatedOutboxAccount | undefined;
+		let terminatedChainOutboxAccount: TerminatedOutboxAccount;
 
 		try {
 			terminatedChainOutboxAccount = await this.stores
 				.get(TerminatedOutboxStore)
-				.get(context, chainIdAsBuffer);
+				.get(context, chainID);
 		} catch (error) {
 			if (!(error instanceof NotFoundError)) {
 				throw error;
 			}
+			return verifyMessageRecovery(context.params);
 		}
 
-		return verifyMessageRecovery(
-			{ idxs, crossChainMessages, siblingHashes },
-			terminatedChainOutboxAccount,
-		);
-	}
-
-	public async execute(context: CommandExecuteContext<MessageRecoveryParams>): Promise<void> {
-		const { params } = context;
+		const deserializedCCMs = crossChainMessages.map(ccm => codec.decode<CCMsg>(ccmSchema, ccm));
 
-		const chainIdAsBuffer = params.chainID;
-
-		const updatedCCMs: Buffer[] = [];
-		const deserializedCCMs = params.crossChainMessages.map(serializedCCMsg =>
-			codec.decode<CCMsg>(ccmSchema, serializedCCMsg),
-		);
 		for (const ccm of deserializedCCMs) {
-			const methodsWithBeforeRecoverCCM = [...this.interoperableCCMethods.values()].filter(method =>
-				Reflect.has(method, 'beforeRecoverCCM'),
-			) as Pick<Required<BaseCCMethod>, 'beforeRecoverCCM'>[];
-			for (const method of methodsWithBeforeRecoverCCM) {
-				await method.beforeRecoverCCM({ ...context, ccm });
+			// The sending chain must be live.
+			const isLive = await this.internalMethod.isLive(context, ccm.sendingChainID, Date.now());
+			if (!isLive) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Cross-chain message sending chain is not live.'),
+				};
 			}
+		}
+
+		return verifyMessageRecovery(context.params, terminatedChainOutboxAccount);
```

Comment:
I don't feel that having this function is much helping us as we only have this code in this verify function and also we are looping twice over the deserialized CCMs just for isLive function. Maybe move all the logic to this place as we are not using it in sidehchain module.